crates/task-impls/src/consensus.rs

```diff
@@ -618,7 +620,7 @@ impl<TYPES: NodeType, I: NodeImplementation<TYPES>, A: ConsensusApi<TYPES, I> +
                 };
                 let Ok(state) = parent_state.validate_and_apply_header(
                     &consensus.instance_state,
-                    &parent_leaf.block_header.clone(),
+                    &parent_leaf.clone(),
```

Comment:
Why are we cloning this if we're passing it in by reference?

Reply:
I don't the cloning is necessary. I'll try removing it.